In [130]:
# Import libaries
import requests
from bs4 import BeautifulSoup
import pandas as pd

### Step 1: Create a soup object from the home page

In [131]:
link = 'https://pages.git.generalassemb.ly/rldaggie/for-scraping/'
page = requests.get(link)
soup = BeautifulSoup(page.content, 'html.parser')

In [132]:
print(soup.prettify())

<!DOCTYPE html>
<html lang="en">
 <head>
  <meta charset="utf-8"/>
  <meta content="IE=edge" http-equiv="X-UA-Compatible"/>
  <meta content="width=device-width, initial-scale=1.0" name="viewport"/>
  <title>
   Nutrition Information
  </title>
  <link crossorigin="anonymous" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" rel="stylesheet"/>
 </head>
 <body>
  <header>
   <section class="container">
    <nav class="navbar navbar-expand-lg navbar-light bg-light" role="navigation">
     <a class="navbar-brand" href="/">
      Nutrition Information
     </a>
    </nav>
   </section>
  </header>
  <main class="container" role="main">
   <br/>
   <div class="alert alert-danger">
    NOTE: This data is super old and rife with errors. It's meant for scraping practice only.
   </div>
   <table class="table" id="restaurants">
    <thead>
     <tr>
      <th>
       Name
      </th>

### Step 2: Scrape the home page soup for every restaurant

Note: Your best bet is to create a list of dictionaries, one for each restaurant. Each dictionary contains the restaurant's name and path from the `href`. The result of your scrape should look something like this:

```python
restaurants = [
    {'name': 'A&W Restaurants', 'href': 'restaurants/1.html'}, 
    {'name': "Applebee's", 'href': 'restaurants/2.html'},
    ...
]
```

In [133]:
restaurants = []

for link in soup.tbody.find_all('a'):
    my_dictionary= {}
    my_dictionary['name'] = link.string
    my_dictionary['href'] = link['href']
    restaurants.append(my_dictionary)

print(restaurants)

[{'name': 'A&W Restaurants', 'href': 'restaurants/1.html'}, {'name': "Applebee's", 'href': 'restaurants/2.html'}, {'name': "Arby's", 'href': 'restaurants/3.html'}, {'name': 'Atlanta Bread Company', 'href': 'restaurants/4.html'}, {'name': "Bojangle's Famous Chicken 'n Biscuits", 'href': 'restaurants/5.html'}, {'name': 'Buffalo Wild Wings', 'href': 'restaurants/6.html'}, {'name': 'Burger King', 'href': 'restaurants/7.html'}, {'name': "Captain D's", 'href': 'restaurants/8.html'}, {'name': "Carl's Jr.", 'href': 'restaurants/9.html'}, {'name': "Charley's Grilled Subs", 'href': 'restaurants/10.html'}, {'name': 'Chick-fil-A', 'href': 'restaurants/11.html'}, {'name': "Chili's", 'href': 'restaurants/12.html'}, {'name': 'Chipotle Mexican Grill', 'href': 'restaurants/13.html'}, {'name': "Church's", 'href': 'restaurants/14.html'}, {'name': 'Corner Bakery Cafe', 'href': 'restaurants/15.html'}, {'name': 'Dairy Queen', 'href': 'restaurants/16.html'}, {'name': "Denny's", 'href': 'restaurants/17.html'}

### Step 3: Using the `href`, scrape each restaurant's page and create a single list of food dictionaries.

Your list of foods should look something like this:
```python
foods = [
    {
        'calories': '0',
        'carbs': '0',
        'category': 'Drinks',
        'fat': '0',
        'name': 'A&W® Diet Root Beer',
        'restaurant': 'A&W Restaurants'
    },
    {
        'calories': '0',
        'carbs': '0',
        'category': 'Drinks',
        'fat': '0',
        'name': 'A&W® Diet Root Beer',
        'restaurant': 'A&W Restaurants'
    },
    ...
]
```

**Note**: Remove extra white space from each category

In [134]:
df = pd.DataFrame()

for item in restaurants:
    link = 'https://pages.git.generalassemb.ly/rldaggie/for-scraping/' + item['href']
    df1 = pd.read_html(link)[0]
    df1['Restaurant'] = item['name']
    df = pd.concat([df,df1],ignore_index=True)

In [135]:
df

Name      Category Calories Fat Carbs  \
0     Original Bacon Double Cheeseburger       Burgers      760  45    45   
1                      Coney (Chili) Dog       Entrees      340  20    26   
2                            Chili Fries  French Fries      370  15    49   
3           Strawberry Milkshake (small)        Shakes      670  29    90   
4          A&W® Root Beer Freeze (large)        Shakes      820  18   150   
...                                  ...           ...      ...  ..   ...   
5126      Jr. Original Chocolate Frosty™        Shakes      200   5    32   
5127             Grilled Chicken Go Wrap         Wraps      260  10    25   
5128           Asiago Ranch Chicken Club    Sandwiches      670  32    57   
5129               Spicy Chicken Go Wrap         Wraps      330  16    30   
5130      Large Strawberry Frosty™ Shake        Shakes      780  15   149   

           Restaurant  
0     A&W Restaurants  
1     A&W Restaurants  
2     A&W Restaurants  
3     A&W Restaurants  
4     A&W Restaurants  
...               ...  
5126          Wendy's  
5127          Wendy's  
5128          Wendy's  
5129          Wendy's  
5130          Wendy's  

[5131 rows x 6 columns]

In [136]:
foods = []
for index in df.index:
    my_dictionary = {}
    my_dictionary['calories'] = df['Calories'][index]
    my_dictionary['carbs'] = df['Carbs'][index]
    my_dictionary['category'] = df['Category'][index]
    my_dictionary['fat'] = df['Fat'][index]
    my_dictionary['Name'] = df['Name'][index]
    my_dictionary['restaurant'] = df['Restaurant'][index]
    foods.append(my_dictionary)

print(foods)

[{'calories': 760, 'carbs': 45, 'category': 'Burgers', 'fat': 45, 'Name': 'Original Bacon Double Cheeseburger', 'restaurant': 'A&W Restaurants'}, {'calories': 340, 'carbs': 26, 'category': 'Entrees', 'fat': 20, 'Name': 'Coney (Chili) Dog', 'restaurant': 'A&W Restaurants'}, {'calories': 370, 'carbs': 49, 'category': 'French Fries', 'fat': 15, 'Name': 'Chili Fries', 'restaurant': 'A&W Restaurants'}, {'calories': 670, 'carbs': 90, 'category': 'Shakes', 'fat': 29, 'Name': 'Strawberry Milkshake (small)', 'restaurant': 'A&W Restaurants'}, {'calories': 820, 'carbs': 150, 'category': 'Shakes', 'fat': 18, 'Name': 'A&W® Root Beer Freeze (large)', 'restaurant': 'A&W Restaurants'}, {'calories': 340, 'carbs': 57, 'category': 'Desserts', 'fat': 9, 'Name': 'Caramel Sundae', 'restaurant': 'A&W Restaurants'}, {'calories': 420, 'carbs': 86, 'category': 'Shakes', 'fat': 6, 'Name': 'Strawberry Banana Smoothee', 'restaurant': 'A&W Restaurants'}, {'calories': 1010, 'carbs': 152, 'category': 'Desserts', 'fat

### Step 4: Create a pandas DataFrame from your list of foods

**Note**: Your DataFrame should have 5,131 rows

In [137]:
df

Name      Category Calories Fat Carbs  \
0     Original Bacon Double Cheeseburger       Burgers      760  45    45   
1                      Coney (Chili) Dog       Entrees      340  20    26   
2                            Chili Fries  French Fries      370  15    49   
3           Strawberry Milkshake (small)        Shakes      670  29    90   
4          A&W® Root Beer Freeze (large)        Shakes      820  18   150   
...                                  ...           ...      ...  ..   ...   
5126      Jr. Original Chocolate Frosty™        Shakes      200   5    32   
5127             Grilled Chicken Go Wrap         Wraps      260  10    25   
5128           Asiago Ranch Chicken Club    Sandwiches      670  32    57   
5129               Spicy Chicken Go Wrap         Wraps      330  16    30   
5130      Large Strawberry Frosty™ Shake        Shakes      780  15   149   

           Restaurant  
0     A&W Restaurants  
1     A&W Restaurants  
2     A&W Restaurants  
3     A&W Restaurants  
4     A&W Restaurants  
...               ...  
5126          Wendy's  
5127          Wendy's  
5128          Wendy's  
5129          Wendy's  
5130          Wendy's  

[5131 rows x 6 columns]

### Step 5: Export to csv

**Note:** Don't export the index column from your DataFrame

In [138]:
df.to_csv('df.csv', index=False)